### PR TITLE
fix: qr code mnemonic

### PR DESCRIPTION
**Pull request by sxiaojian88**

### CODE REVIEW

File: Maskbook/Extension/Data+Base64URL.swift [hunk 1]
```diff
@@ -52,6 +52,10 @@ extension String {
             .replacingOccurrences(of: "/", with: "_")
             .replacingOccurrences(of: "=", with: "")
     }
+    
+    public func urlEncode() -> String? {
```
Comment (foxsin10) [inline Maskbook/Extension/Data+Base64URL.swift]: Recommend using`percentEncoding(using filterSet: CharacterSet = CharacterSet(charactersIn: "!*'();:@&=+$,/?%#[]\" {}^|")) -> String?` instead.

File: Maskbook/Service/SchemeService/SchemeService.swift [hunk 2]
```diff
@@ -71,6 +84,30 @@ class SchemeService {
         let personaImportItem = PersonaImportItem(type: .mnemonic(mnemonic: mnemonic), name: nickname)
         personaImportHandler.checkExistAndRestore(from: personaImportItem)
     }
+    
+    func personaMnemonicURLEscaped(string: String) -> String {
```
Comment (foxsin10) [inline Maskbook/Service/SchemeService/SchemeService.swift:88]: Better add some comments for detail of this function

Comment (foxsin10) [inline Maskbook/Service/SchemeService/SchemeService.swift:92]: Better add some comments for detail of this function.

File: Maskbook/Extension/Data+Base64URL.swift (hunk 1 above)
Comment (sxiaojian88) [reply Maskbook/Extension/Data+Base64URL.swift:56]: Why？

File: Maskbook/Service/SchemeService/SchemeService.swift (hunk 2 above)
Comment (sxiaojian88) [reply Maskbook/Service/SchemeService/SchemeService.swift:88]: ok

File: Maskbook/Extension/Data+Base64URL.swift (hunk 1 above)
Comment (foxsin10) [reply Maskbook/Extension/Data+Base64URL.swift]: This is a function for `String` extension, and should has a specific name to indicate what it does, not just `urlEncode`.

Comment (sxiaojian88) [reply Maskbook/Extension/Data+Base64URL.swift:56]: It's purpose is just urlEncode, I don't want to expose a complex parameter to caller.
And other parameter does another process, that is not my purpose too.
The file `SocketExtensions.swift` do the same.

Comment (foxsin10) [reply Maskbook/Extension/Data+Base64URL.swift]: Then rename to `usingpercentEncoding` will be fine.

File: Maskbook/Service/SchemeService/SchemeService.swift [hunk 3]
```diff
@@ -71,6 +84,34 @@ class SchemeService {
...
+    
+    
+    /// process url to URLEncoded string to avoid URL(string: string) be nil
+    /// - Parameter string: url
+    /// - Returns: URLEncoded string
+    func personaMnemonicURLEscaped(string: String) -> String {
+        var nicknameString: String?
+        let prefix: String? = {
+            if string.contains("?") {
+                let array = string.split(separator: "?").map({String($0)})
```
Comment (foxsin10) [inline Maskbook/Service/SchemeService/SchemeService.swift]: May be this code block will be more efficient.
```
let slices = string.split(separator: "?")
nicknameString = slices.last?.split(separator: "?").last.flatMap { String($0) }
return slices.first.flatMap { String($0) }
```

**Review by foxsin10 — APPROVED** [review]

**Pull request merged**